go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -271,12 +274,31 @@ func (qre *QueryExecutor) checkPermissions() error {
 	}
 
 	// empty table name, do not need a table ACL check.
-	if qre.plan.TableName().IsEmpty() {
+	if qre.plan.TableName().IsEmpty() && qre.plan.NewName != nil && qre.plan.NewName.IsEmpty() {
 		return nil
 	}
-
+	if qre.plan.NewName != nil && !qre.plan.NewName.IsEmpty() {
+		altAuthorized := tableacl.Authorized(qre.plan.NewName.String(), qre.plan.PlanID.MinRole())
```

Comment:
Can we just pull out the code from line 303-324 into a function, and call it with NewName and TableName separately? This looks like a good amount of mostly duplicate code.